shared/actions/chat2/index.js

```diff
@@ -1792,7 +1792,7 @@ function* chat2Saga(): Saga.SagaGenerator<any, any> {
   }
 
   // Refresh the inbox
-  yield Saga.safeTakeEveryPure([Chat2Gen.inboxRefresh, Chat2Gen.leaveConversation], inboxRefresh)
```

Comment:
i think we need to understand why this doesn't work

Reply:
So turns out a race exists in this flow:

1. We get a bad version of the inbox and start trying to unbox them in `unboxRows` (because we have opened the inbox with the bad conversation in view (?))
2. In the meantime, we get a good version of the inbox, clear out the `metaMap`, and refill it with all the valid conversations we're members of
3. The `onUnboxed` callback returns for the bad item in the first inbox call and we re-insert it into `metaMap`, leading to the bug seen here.

In this case we can avoid the race completely by counting on the service to tell us when to refresh. I'm documenting it here because this could be a source of bugs in the future.

Reply:
i'm fixing this race now

Reply:
https://github.com/keybase/client/pull/11075